tests/crypto/test_cases/test_aead.c

```diff
@@ -0,0 +1,575 @@
+	TEST_VECTOR_ASSERT_EQUAL(p_test_vector->expected_err_code, err_code);
+
+	TEST_VECTOR_MEMCMP_ASSERT(m_aead_expected_output_buf, m_aead_output_buf, output_len, p_test_vector->crypt_expected_result, "Output buf check");
+
+	/* Overflow checks */
+	TEST_VECTOR_OVERFLOW_ASSERT(m_aead_output_buf, output_len, "output buffer overflow");
+	TEST_VECTOR_OVERFLOW_ASSERT(m_aead_output_mac_buf, mac_len, "MAC buffer overflow");
+
+	/* Free resources. */
+	mbedtls_cipher_free(&ctx);
+	g_encrypt = true;
+}
+
+/**@brief Function for the AEAD test execution.
+ */
+void exec_test_case_aead_simple(void)
+{
+	int err_code = -1;
+
+	#if defined(MBEDTLS_CCM_C)
+	if (p_test_vector->mode == MBEDTLS_MODE_CCM && p_test_vector->ccm_star) {
+		exec_test_case_aead_ccm_star_simple();
+		return;
+	}
+	#endif
+
+	/* Initialize AEAD. */
+	key_bits = key_len * 8;
+
+	mbedtls_cipher_context_t ctx;
+	mbedtls_cipher_init(&ctx);
+
+	/* Setup cipher. */
+	const mbedtls_cipher_info_t *p_info = mbedtls_cipher_info_from_values(p_test_vector->id, key_bits, p_test_vector->mode);
+
+	err_code = mbedtls_cipher_setup(&ctx, p_info);
+	LOG_DBG("Err code setup: -0x%04X", -err_code);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	/* Set secret key. */
+	err_code = mbedtls_cipher_setkey(&ctx, m_aead_key_buf, key_bits, p_test_vector->direction);
+	LOG_DBG("Err code setkey: -0x%04X", -err_code);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	size_t operation_len = output_len;
+	start_time_measurement();
+	if (p_test_vector->direction == MBEDTLS_ENCRYPT) {
+		err_code = mbedtls_cipher_auth_encrypt(&ctx,
+						       m_aead_nonce_buf,
+						       nonce_len,
+						       m_aead_ad_buf,
+						       ad_len,
+						       m_aead_input_buf,
+						       input_len,
+						       m_aead_output_buf,
+						       &operation_len,
+						       m_aead_output_mac_buf,
+						       mac_len);
+	} else {
+		err_code = mbedtls_cipher_auth_decrypt(&ctx,
+						       m_aead_nonce_buf,
+						       nonce_len,
+						       m_aead_ad_buf,
+						       ad_len,
+						       m_aead_input_buf,
+						       input_len,
+						       m_aead_output_buf,
+						       &operation_len,
+						       m_aead_output_mac_buf,
+						       mac_len);
+	}
+
+	stop_time_measurement();
+
+	LOG_DBG("Err code %s: -0x%04X, operation len: %d", p_test_vector->direction ? "encrypt" : "decrypt", -err_code, operation_len);
+	TEST_VECTOR_ASSERT_EQUAL(p_test_vector->expected_err_code, err_code);
+
+	if (input_len != 0) {
+		TEST_VECTOR_MEMCMP_ASSERT(m_aead_expected_output_buf, m_aead_output_buf, output_len, p_test_vector->crypt_expected_result, "Output buf check");
+	}
+	if (p_test_vector->direction == MBEDTLS_ENCRYPT) {
+		TEST_VECTOR_MEMCMP_ASSERT(m_aead_expected_mac_buf, m_aead_output_mac_buf, mac_len, p_test_vector->mac_expected_result, "MAC buf check");
+	}
+
+	/* Free resources. */
+	mbedtls_cipher_free(&ctx);
+}
+
+/* Simple wrapping functions redirecting the various cases to common functionality. */
```

Comment:
Why? Why not just register exec_test_case_aead directly?

Reply:
This was related to host test logic.
Removed in b2a32c9